From 7342bdd72094a89d49c3dc00196ee7c7532c4b83 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:08:10 -0400
Subject: [usb] qcserial: remove interface number matching

Message-id: <1410887463-20674-419-git-send-email-dzickus@redhat.com>
Patchwork-id: 95450
O-Subject: [RHEL7 PATCH 418/591] usb: qcserial: remove interface number matching
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 48292d8b0726412646086821656193dbc289ce4c
Author: Bjorn Mork <bjorn@mork.no>
Date:   Sun Apr 27 16:47:46 2014 +0200

    usb: qcserial: remove interface number matching

    Matching on interface numbers was not such a good idea
    for multi-function serial devices after all. It is much
    better do create well defined device layouts, allowing
    a single match entry per device.

    Remove this now unused code.

    Signed-off-by: Bjorn Mork <bjorn@mork.no>
    Signed-off-by: Johan Hovold <jhovold@gmail.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/serial/qcserial.c b/drivers/usb/serial/qcserial.c
index 91e7bb5..ca7b430 100644
--- a/drivers/usb/serial/qcserial.c
+++ b/drivers/usb/serial/qcserial.c
@@ -190,13 +190,6 @@ static int qcprobe(struct usb_serial *serial, const struct usb_device_id *id)
 
  }
 
- /* allow any number of interfaces when doing direct interface match */
- if (id->match_flags & USB_DEVICE_ID_MATCH_INT_NUMBER) {
-  dev_dbg(dev, "Generic Qualcomm serial interface found\n");
-  altsetting = 0;
-  goto done;
- }
-
  /* default to enabling interface */
  altsetting = 0;
 
-- 
1.7.1